pkg/certificate/auto/provider.go

```diff
@@ -0,0 +1,314 @@
+package auto
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"time"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	interfaces "go.etcd.io/etcd-operator/pkg/certificate/interfaces"
+	"go.etcd.io/etcd/client/pkg/v3/transport"
+)
+
+const (
+	DefaultValidity = 365 * 24 * time.Hour
+)
+
+type Provider struct {
+	client.Client
+}
+
+var _ interfaces.Provider = (*Provider)(nil)
+
+func New(c client.Client) interfaces.Provider {
+	return &Provider{
+		c,
+	}
+}
+
+var userAutoConfig *interfaces.Config
```

Comment:
Why define a global variable? Please add it as a field in Provider struct (line 34 above).

Reply:
pls also resolve this in a followup PR, thx